### PR TITLE
Docs: Styling edits for Deploying to Openshift guide

### PR DESCRIPTION
**Pull request by sheilamjones**
This PR aims to enhance the existing Deploying on OpenShift guide as follows:

- Complete iteration 1 of restructure of Deploying on OpenShift guide into Diataxis content types. Restructure updates include the following changes:
  - Introduce a new concept topic "deployment-to-openshift.adoc" to provide a overview of deployment to OpenShift concepts
  - Introduce 2 new "How-to" topics based on downstream Deploy to OpenShift guide content. The new topics are:
  -  Deploy Quarkus applications as an OpenShift Serverless service ("deploying-to-openshift-serverless.adoc")
  -  Deploy a Quarkus native application on OpenShift ("deploying-native-executable-openshift.adoc")

- Edit to enhance and style the existing guide as per the [Quarkus doc contributor guide](https://quarkus.io/guides/doc-reference)

- Implement corrections arising from Vale Linter checks

### CODE REVIEW

**Comment by sheilamjones** [comment]
Hi @iocanel, @mkouba, @Ladicek,
Please see draft updates to the Deploying to OpenShift guide for which I would appreciate your review. Per our planning discussion a couple of weeks ago, I completed styling edits on the existing guide per style and Diataxis guidelines. I created a concept and new How To topics to backport corresponding Deploy to OpenShift content from downstream.  
Many thanks in advance,
Sheila

**Review by iocanel — CHANGES_REQUESTED** [review]
Great work overall.

I started adding some comments to the files in the order they appeared in the review tab only to realize that lots of them are addressed by `deployment-to-openshift.adoc` that I assume is meant to be read first.

To be more specific the individual files felt like they are missing an overview that was presented in the `deployment-to-openshift.adoc`. That being said we can't be really sure how the user may land to each of this pages, so maybe a smaller overview (possibly pointing to `deployment-to-openshift.adoc` is needed.

Other than that added some minor comments here and there.

**Comment by sheilamjones** [comment]
Hi @iocanel, many thanks for your review and feedback. Working to process these today. With regards to your commend about the overview and conceptual information, you are correct. The intention was to incorporate that information in the "deployment-to-openshift" conceptual topic and provide links from there to the individual How Tos. But, I could enhance the overviews of the individual How Tos per your suggestion and perhaps link to the larger concept topic at the beginning also.

File: docs/src/main/asciidoc/deploying-native-executable-openshift.adoc [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+[source,bash]
+----
+quarkus.openshift.build-strategy=docker
+----
+
+.. Set the container runtime:
++
+[source,bash]
+----
+quarkus.native.container-build=true
```
[the remark this answers is not in the document]
Comment (sheilamjones) [reply docs/src/main/asciidoc/deploying-native-executable-openshift.adoc:119]: Updated to add "Optional" prefix to the introductory statement:

.. Optional: Set the container runtime:
+
[source,bash]
----
quarkus.native.container-build=true
----

File: docs/src/main/asciidoc/deployment-to-openshift.adoc [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+ You can build and deploy your Quarkus applications to OpenShift Container Platform in several ways:
+
+* By using the `quarkus-openshift` extension
++
+The OpenShift extension is a wrapper extension that brings together the xref:deploying-to-kubernetes.adoc[kubernetes] and xref:container-image.adoc#openshift[container-image-openshift] extensions with sensible default values, which help you to get started with Quarkus on OpenShift.
+The extension supports multiple deployment options, including an OpenShift binary build strategy or Docker build strategy.
+
+* By using the Quarkus CLI
++
+You can issue the `quarkus deploy` command to enable the deployment of Quarkus applications to platforms, such as Kubernetes or OpenShift Container Platform.
```
[the remark this answers is not in the document]
Comment (sheilamjones) [reply docs/src/main/asciidoc/deployment-to-openshift.adoc:28]: Thank you. Updated and link also provided to https://quarkus.io/guides/cli-tooling guide for more detailed information.

File: docs/src/main/asciidoc/deploying-native-executable-openshift.adoc [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+----
+
+[TIP]
+====
+You can request the token by using the _Copy Login Command_ link in the OpenShift web console.
+====
+
+To log in by setting the `quarkus.kubernetes-client.api-server-url` configuration property and authenticating with the `quarkus.kubernetes-client.token`, or `quarkus.kubernetes-client.username` and `quarkus.kubernetes-client.password` respectively:
+
+:build-additional-parameters: -Dquarkus.kubernetes-client.api-server-url=myServerUrl -Dquarkus.kubernetes-client.token=myToken
```
[the remark this answers is not in the document]
Comment (sheilamjones) [reply docs/src/main/asciidoc/deploying-native-executable-openshift.adoc:84]: Thank @iocanel. Can I clarify exactly what I need to change or advise what the command should for the QUARKUS_KUBERNETES_CLIENT_TOKEN environment variable alternative?

File: docs/src/main/asciidoc/deploying-native-executable-openshift.adoc [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+////
+[id="deploy-openshift-native-executable"]
+= Deploy a Quarkus native application on OpenShift
+include::_attributes.adoc[]
+:diataxis-type: howto
+:categories: cloud,native
+
+Deploy native Quarkus applications to OpenShift Container Platform by using the Docker build strategy enabled by the Quarkus OpenShift extension.
+
+Quarkus supports the `quarkus-container-image-docker` extension for building container images.
```
Comment (iocanel) [inline docs/src/main/asciidoc/deploying-native-executable-openshift.adoc:14]: I think that this paragraph is not accurate.
The docker build strategy enabled by the openshift extension has nothing to do with `quarkus-container-image-docker` it's a feature provided by `quarkus-container-image-openshift` that is automatically brought into the project when users add `quarkus-openshift` to their project.

File: docs/src/main/asciidoc/deploying-native-executable-openshift.adoc [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+The extension uses the Docker binary and generated Dockerfiles in the `src/main/docker` directory to perform Docker builds.
```
Comment (iocanel) [inline docs/src/main/asciidoc/deploying-native-executable-openshift.adoc:15]: The extension (meaning the quarkus-container-image-openshift) is not using the docker binary. It is uploading the native binary and the dockerfiles to openshift and openshift performs the build.

In fact this could all work without the user having docker installed. I am saying `could` because in some cases the `docker` binary may be used in order to create the native binary itself (as correctly described by the paragraph below).

File: docs/src/main/asciidoc/deployment-to-openshift.adoc [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+[source,bash]
+----
+quarkus.container-image.builder=docker
+----
+
+The Docker build strategy builds the artifacts outside the OpenShift Container Platform cluster, locally or in a continuous integration (CI) environment, and provides them to the OpenShift Container Platform build system together with a Dockerfile.
+The artifacts include JAR files or a native executable.
+The container gets built inside the OpenShift Container Platform cluster and is provided as an image stream.
+
+== Build strategies supported by Quarkus
```
Comment (iocanel) [inline docs/src/main/asciidoc/deployment-to-openshift.adoc:80]: I think that this table is confusing and maybe we are better off without it.

**Review by iocanel — CHANGES_REQUESTED** [review]
Did a second pass and resolved most of my previous comments that have been addressed.
Add a new one regarding the mention to `quarkus-container-image-docker` that I missed in my first pass.

File: docs/src/main/asciidoc/deploying-native-executable-openshift.adoc (hunk 4 above)
Comment (sheilamjones) [reply docs/src/main/asciidoc/deploying-native-executable-openshift.adoc:14]: Thank you. Deleted the line "Quarkus supports the `quarkus-container-image-docker` extension for building container images."

File: docs/src/main/asciidoc/deploying-native-executable-openshift.adoc (hunk 5 above)
Comment (sheilamjones) [reply docs/src/main/asciidoc/deploying-native-executable-openshift.adoc:15]: Would the following update be correct instead:  "The `quarkus-container-image-openshift` extension uploads a native binary and Dockerfiles to OpenShift and OpenShift performs the Docker builds."

File: docs/src/main/asciidoc/deployment-to-openshift.adoc (hunk 6 above)
Comment (sheilamjones) [reply docs/src/main/asciidoc/deployment-to-openshift.adoc]: Removed table.

File: docs/src/main/asciidoc/deploying-to-openshift.adoc [hunk 7]
```diff
@@ -161,48 +193,120 @@ oc expose svc/greeting <3>
...
 <2> Create a new application from the image source.
 <3> Expose the service to the outside world.
 <4> Get the list of exposed routes.
 <5> Access your application.
 
-After this setup the next time the container image is built a deployment to OpenShift is triggered automatically.
-In other words, you don't need to repeat the above steps.
+After this setup, the next time you build a container image, a deployment to OpenShift is triggered automatically so you do not need to repeat the above steps.
+
+=== Option: Configure OpenShift application manually by using the Docker build
```
Comment (sheilamjones) [inline docs/src/main/asciidoc/deploying-to-openshift.adoc:205]: Hi @iocanel, Thank you for your additional review and I updated the sections per your feedback. Following our meeting, I also updated the _Choose how you want to build and deploy your Quarkus application_ chapter to add the Docker build strategy as a subsection (content backported from downstream guide). Does this look okay to you?

**Comment by sheilamjones** [comment]
Hi @iocanel, 
Thank you for your additional review and I updated the sections per your feedback. One outstanding section is the building using the docker build strategy, which I added following our meeting. 

@maxandersen, @tqvarnst, @cescoffier, @mkouba - I would appreciate your review of these updates also. The changes comprise a new concept topic, and the existing Openshift guide divided to comprise 3 how to guides (deploying to Openshift Container Platform, deploying to serverless, deploying native executable)

Many thanks,
Sheila

**Review by iocanel — APPROVED** [review]
I marked all my comments as resolved.
All? There is still one unresolved about the need of `docker` binary for the openshift docker build strategy. I confirm that the suggested change sounds good and once that's in all my concerns will be addressed.

File: docs/src/main/asciidoc/deploying-to-openshift.adoc [hunk 8]
```diff
@@ -49,70 +57,90 @@ When we added the OpenShift extension to the command line invocation above, the
...
+[NOTE]
+====
+When you use `DeploymentConfig` and link:https://quarkus.io/guides/deploying-to-kubernetes#service_binding[Service Binding], re-deploying might remove the configuration that is added by OpenShift to allow service discovery.
 
-This command will build your application locally, then trigger a container image build and finally apply the generated OpenShift resources automatically.
-The generated resources use OpenShift's `DeploymentConfig` that is configured to automatically trigger a redeployment when a change in the `ImageStream` is noticed.
-In other words, any container image build after the initial deployment will automatically trigger redeployment, without the need to delete, update or re-apply the generated resources.
+A new container image build triggers a refresh of the Quarkus application in OpenShift, `-Dquarkus.container-image.build=true`, which might suffice in most situations.
+If you must update the OpenShift resources, you first delete the binding to create it again after a new deployment.
```
[the remark this answers is not in the document]
Comment (cescoffier) [reply docs/src/main/asciidoc/deploying-to-openshift.adoc:125]: (First, the reader is probably not aware of what is service binding)

File: docs/src/main/asciidoc/deploying-to-openshift.adoc [hunk 9]
```diff
@@ -49,70 +57,90 @@ When we added the OpenShift extension to the command line invocation above, the
...
-<1> Lists the image streams created.
-<2> Get the list of pods.
+<1> List the image streams created.
+<2> View a list of pods associated with your current OpenShift project.
 <3> Get the list of Kubernetes services.
 
-Note that the service is not exposed to the outside world by default.
-So unless you've used the `quarkus.openshift.route.expose` config property to expose the created service automatically you'll need to expose the service manually.
+The service is not exposed to the outside world by default.
+Therefore, unless you have used the `quarkus.openshift.route.expose` configuration property to expose the created service automatically, you must expose the service manually.
```
Comment (cescoffier) [inline docs/src/main/asciidoc/deploying-to-openshift.adoc:141]: That was mentioned before, I would remove it from above to mention is in a single place.

File: docs/src/main/asciidoc/deploying-to-openshift.adoc [hunk 10]
```diff
@@ -161,48 +193,120 @@ oc expose svc/greeting <3>
...
 <5> Access your application.
 
-After this setup the next time the container image is built a deployment to OpenShift is triggered automatically.
-In other words, you don't need to repeat the above steps.
+After this setup, the next time you build a container image, a deployment to OpenShift is triggered automatically so you do not need to repeat the above steps.
+
+=== Option: Configure OpenShift application manually by using the Docker build
 
-=== Non-S2I Builds
+Build and deploy your Quarkus application to OpenShift Container Platform by using the Docker build strategy.
```
Comment (cescoffier) [inline docs/src/main/asciidoc/deploying-to-openshift.adoc:207]: I would just reference the other guide.

File: docs/src/main/asciidoc/deploying-to-openshift.adoc [hunk 11]
```diff
@@ -161,48 +193,120 @@ oc expose svc/greeting <3>
...
 - xref:container-image.adoc#jib[container-image-jib]
 
-When a non-s2i container image extension is used, an `ImageStream` is created that is pointing to an external `dockerImageRepository`. The image is built and pushed to the registry and the `ImageStream` populates the tags that are available in the `dockerImageRepository`.
+When you use a non-OpenShift container image extension, an `ImageStream` is created that points to an external `dockerImageRepository`.
+The image is built and pushed to the registry and the `ImageStream` populates the tags that are available in the `dockerImageRepository`.
 
-To select which extension will be used for building the image:
+Select the extension you want to use to build the image, by choosing one of the following commands.
 
+* To use the `quarkus-container-image-docker` extenstion, run the following command:
```
Comment (cescoffier) [inline docs/src/main/asciidoc/deploying-to-openshift.adoc:281]:
```suggestion
* To use the `quarkus-container-image-docker` extension, run the following command:
```

File: docs/src/main/asciidoc/deploying-to-openshift.adoc [hunk 12]
```diff
@@ -161,48 +193,120 @@ oc expose svc/greeting <3>
...
-All available customization options are available in the xref:deploying-to-kubernetes.adoc#openshift[OpenShift configuration options].
+== Customize your deployment
+
+For a list of available customization options, see xref:deploying-to-kubernetes.adoc#openshift[OpenShift configuration options].
 
 Some examples are provided in the sections below:
 
 [[exposing_routes]]
-=== Exposing Routes
+=== Expose routes
```
Comment (cescoffier) [inline docs/src/main/asciidoc/deploying-to-openshift.adoc:307]: REference content - should not be there

File: docs/src/main/asciidoc/deploying-to-openshift.adoc [hunk 13]
```diff
@@ -221,11 +327,12 @@ You don't necessarily need to add this property in the `application.properties`.
 The same applies to all properties listed below.
 ====
 
-==== Securing the Route resource
+==== Secure the Route resource
```
Comment (cescoffier) [inline docs/src/main/asciidoc/deploying-to-openshift.adoc:330]: Reference content - should not be there

File: docs/src/main/asciidoc/deploying-to-openshift.adoc [hunk 14]
```diff
@@ -236,57 +343,57 @@ quarkus.openshift.route.tls.termination=passthrough
 quarkus.openshift.route.tls.insecure-edge-termination-policy=None
 ----
 
-=== Labels
+=== Add labels
```
Comment (cescoffier) [inline docs/src/main/asciidoc/deploying-to-openshift.adoc:346]: Reference content - should not be there

File: docs/src/main/asciidoc/deploying-to-openshift.adoc [hunk 15]
```diff
@@ -236,57 +343,57 @@ quarkus.openshift.route.tls.termination=passthrough
...
-To add a label in the generated resources:
+To add a label in the generated resources, run the following command:
 
 [source,properties]
 ----
 quarkus.openshift.labels.foo=bar
 ----
 
-=== Annotations
+=== Add annotations
```
Comment (cescoffier) [inline docs/src/main/asciidoc/deploying-to-openshift.adoc:355]: Reference content - should not be there

File: docs/src/main/asciidoc/deployment-to-openshift.adoc [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+:diataxis-type: concept
+:categories: cloud,native
+
+Quarkus supports multiple deployment options to deploy Quarkus applications to OpenShift Container Platform.
+
+== Quarkus deployment to OpenShift overview
+
+OpenShift Container Platform is a Kubernetes-based platform for developing and running containerized applications.
+As an application developer, to deploy Quarkus applications OpenShift Container Platform, you first choose how you want to build the application, for example, by using the `quarkus-openshift` extension or the Quarkus CLI.
+Then, you choose how you want to build the container image.
```
[the remark this answers is not in the document]
Comment (cescoffier) [reply docs/src/main/asciidoc/deployment-to-openshift.adoc:18]: I would say that a flow chart would be super nice!

File: docs/src/main/asciidoc/deployment-to-openshift.adoc [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+- xref:container-image.adoc#jib[Container-image-jib]
+- xref:container-image.adoc#buildpack[Container-image-buildpack]
+
+[NOTE]
+====
+The OpenShift Container Platform Docker build strategy is the preferred non-OpenShift build strategy because it supports Quarkus applications targeted for JVM or compiled to native executables.
+However, for compatibility with earlier Quarkus versions, the default build strategy is the OpenShift binary build.
+To select the OpenShift Container Platform Docker build strategy, use the `quarkus.openshift.build-strategy` property.
+====
+
```
Comment (cescoffier) [inline docs/src/main/asciidoc/deployment-to-openshift.adoc:66]: Something is missing here.
The extensions mentioned above are creating the container, then you need to push it.

File: docs/src/main/asciidoc/deployment-to-openshift.adoc [hunk 18]
```diff
@@ -0,0 +1,98 @@
...
+If you use a non-OpenShift container image extension, an `ImageStream` is created that points to an external `dockerImageRepository`.
```
Comment (cescoffier) [inline docs/src/main/asciidoc/deployment-to-openshift.adoc:67]: link image stream - openshift specific resource.

**Review by cescoffier — CHANGES_REQUESTED** [review]

**Comment by cescoffier** [comment]
Alright,

I think we need to split that PR

1. Update to the openshift guide should be done first in a separate PR
2. The new deployment of native to openshift in another separate PR
3. The knative one should be separated too
4. The concept guide (which require a bit of rework) should come last

File: docs/src/main/asciidoc/deploying-to-openshift.adoc [hunk 19]
```diff
@@ -4,35 +4,43 @@ and pull requests should be submitted there:
...
 
 == Prerequisites
 
 :prerequisites-no-graalvm:
 include::{includes}/prerequisites.adoc[]
-* Access to an OpenShift cluster (Minishift is a viable option)
-* OpenShift CLI (Optional, only required for manual deployment)
+* Access to an OpenShift Container Platform cluster and the latest compatible version of the `oc` tool installed (Minishift is a viable option)
+* Access to the link:https://docs.openshift.com/container-platform/4.9/cli_reference/openshift_cli/getting-started-cli.html[OpenShift CLI] (Optional, only required for manual deployment)
+* Mandrel or GraalVM installed and configured appropriately.
```
[the remark this answers is not in the document]
Comment (sheilamjones) [reply docs/src/main/asciidoc/deploying-to-openshift.adoc:23]: Replaced "Mandrel or GraalVM installed and configured appropriately" with "Docker or Podman installed and configured appropriately"

File: docs/src/main/asciidoc/deploying-to-openshift.adoc [hunk 20]
```diff
@@ -4,35 +4,43 @@ and pull requests should be submitted there:
...
-== Bootstrapping the project
+:sectnums:
+:sectnumlevels: 3
 
-First, we need a new project that contains the OpenShift extension. This can be done using the following command:
+== Create a new project that includes the Quarkus OpenShift extension
+
+Create a new project that includes the `quarkus-openshift` extension.
+To build and deploy your applications as a container image that runs inside your OpenShift Container Platform cluster, add the Quarkus OpenShift extension as a dependency to your project.
+Run the following command:
```
[the remark this answers is not in the document]
Comment (sheilamjones) [reply docs/src/main/asciidoc/deploying-to-openshift.adoc:33]: The first sentence was an introduction. I removed it. The following 2 sentences are the instruction and the command needed to add the extension as the dependency.

**Comment by sheilamjones** [comment]
Update:
Closing this PR per advice to split this PR into smaller separate PRs that would cover each guide separately. Feedback and discussion from this PR will be transferred to the new PRs.
- PR: https://github.com/quarkusio/quarkus/pull/34903 - Deploying to OpenShift How To plus Reference guides
- PR: https://github.com/quarkusio/quarkus/pull/35268 - Deploy native applications to OpenShift
- PR: Deploy to Serverless and Deployment to OpenShift concept (to be added)
